x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/CapturingActionListener.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.security.support;
+
+import org.elasticsearch.action.ActionListener;
+
+/**
+ * Simple implementation of ActionListener to capture response or exception.
+ * Allowed a caller to pass this listener and get a response.
+ * @param <T> Response class
+ */
+public class CapturingActionListener<T> implements ActionListener<T> {
```

Comment:
You can use `PlainActionFuture` in tests instead of needing to add this new class.

Reply:
Note, a similar CapturingActionListener class is used elsewhere in Elasticsearch. I use it in tests to capture the exception and get it, without having the underlying call rethrow the exception. That allows for assertions on the exception message, if necessary.

If it is OK, I will defer this to the next PR.